src/lib.rs

```diff
@@ -1065,7 +1101,11 @@ impl<T: Activated + ?Sized> Capture<T> {
     /// Blocks until a packet is returned from the capture handle or an error occurs.
     ///
     /// pcap captures packets and places them into a buffer which this function reads
-    /// from. This buffer has a finite length, so if the buffer fills completely new
+    /// from.
+    ///
+    /// # Warning
+    ///
+    /// This buffer has a finite length, so if the buffer fills completely new
     /// packets will be discarded temporarily. This means that in realtime situations,
     /// you probably want to minimize the time between calls of this next() method.
     #[allow(clippy::should_implement_trait)]
```

Comment:
Can we remove this clippy `allow` then? Isn't the whole point of this PR to get rid of it?

Reply:
no really, we would disallow a number of usecase, we could rename it however.

Reply:
Yes, let's rename it. I propose `next_packet` to align with the raw libpcap call and inidcate that it's like an iterator, but it is not one. However, I don't have any strong opinions on it. As long as it's not `next` as it makes the iterator's `next` implementation slightly confusing.